### DISCUSSION TITLE
Code Generation Fails When Binding Types: "Unable to Bind to Interface

### DISCUSSION THREAD

**Discussion by islamaliev** [99designs/gqlgen#2835] [General]
Hello gqlgen community,

I'm facing a perplexing issue with gqlgen's code generation in my Go project. I have a GraphQL schema and a Go interface, and I'm trying to bind a GraphQL type `CourseProduct` to a Go struct. However, I'm encountering an error during the code generation process that I can't seem to resolve.

GraphQL Schema:
```graphql
interface Product @goModel(
    model: "project.com/internal/service/product/model.Product"
) {
  id: Int!
}

type CourseProduct implements Product @goModel(
    model: "project.com/internal/service.CourseWrapper2"
) {
  id: Int!
  title: String!
}
```

I have a Go interface defined in `internal/service/product/model/product.go`:

```go
package model

import (
    "time"
)

type Product interface {
    GetID() int
    GetTitle() string
    GetCreatedAt() time.Time
}
```

I also have wrapper structs in different packages:

`CourseWrapper1` in `internal/service/product`
`CourseWrapper2` in `internal/service`
Both wrappers embed `*model.Course`.

When I run `go generate ./...`, the code generation fails with the following error:

```
merging type systems failed: unable to bind to interface: project.com/internal/service.CourseWrapper2 does not satisfy the interface project.com/internal/service/product/model.Product
exit status 3
resolver.go:17: running "go": exit status 1
generate: open /path/to/project/api/graphql/generated/generated.go: no such file or directory
```

Interestingly, if I bind `CourseProduct` to `project.com/internal/service/product.CourseWrapper1` (which is essentially the same struct but one package closer to `model.Product`), the error disappears.

Additional Observation:
If I comment out the `GetCreatedAt() time.Time` method in the `model.Product` interface, the code generates successfully even when binding to `project.com/internal/service.CourseWrapper2`.

I've tried debugging this issue, but it goes deep into `go/types`, `golang.org/x/tools/go/packages/packages.go`, and other internals of how Go loads and resolves packages and types.

I've also posted this question on [Stack Overflow](https://stackoverflow.com/questions/77362098/gqlgen-code-generation-fails-when-binding-types-unable-to-bind-to-interface) for additional visibility.

Any insights or suggestions would be greatly appreciated!

**Comment by pwaterz** [comment]
You need to implement the interface methods.

Example I had this interface 

```
interface Node {
    id: ID!
}
```

I had to add this to my object
```
func (ObjectNmae) IsNode() {}
```
